service/src/main/kotlin/io/provenance/api/domain/usecase/cee/execute/ExecuteContractBatch.kt

```diff
@@ -0,0 +1,95 @@
+    private val contractUtilities: ContractUtilities,
+) : AbstractUseCase<ExecuteContractBatchRequestWrapper, List<ContractExecutionResponse>>() {
+
+    private val log = KotlinLogging.logger { }
+
+    override suspend fun execute(args: ExecuteContractBatchRequestWrapper): List<ContractExecutionResponse> {
+        val responses = mutableListOf<ContractExecutionResponse>()
+        val errors = mutableListOf<Throwable>()
+        val results = mutableListOf<ExecutionResult>()
+        val signer = getSigner.execute(GetSignerRequest(args.uuid, args.request.config.account))
+        val client = contractUtilities.createClient(args.uuid, args.request.permissions, args.request.participants, args.request.config)
+
+        contractUtilities.createSession(args.uuid, args.request.permissions, args.request.participants, args.request.config, args.request.records, args.request.scopes).forEach {
+            results.add(contractService.executeContract(client, it))
+        }
+
+        val chunkedSignedResult = results.filterIsInstance(SignedResult::class.java).chunked(args.request.chunkSize)
+        chunkedSignedResult.forEachIndexed { index, it ->
+            runCatching {
+                provenanceService.buildContractTx(args.request.config.provenanceConfig, BatchTx(it)).let { tx ->
+                    provenanceService.executeTransaction(args.request.config.provenanceConfig, tx, signer).let { pbResponse ->
+                        responses.add(
+                            ContractExecutionResponse(
+                                false,
+                                null,
+                                pbResponse.toTxResponse()
+                            )
+                        )
+                    }
+                }
+            }.fold(
+                onSuccess = {
+                    log.info("Successfully processed batch $index of ${chunkedSignedResult.size}")
+                },
+                onFailure = {
+                    errors.add(it)
+                }
+            )
+        }
+
+        val chunkedFragResult = results.filterIsInstance(FragmentResult::class.java).chunked(args.request.chunkSize)
+        chunkedFragResult.forEachIndexed { index, chunk ->
+            runCatching {
+                chunk.forEach { result ->
+                    client.requestAffiliateExecution(result.envelopeState)
+                    responses.add(
+                        ContractExecutionResponse(
+                            true,
+                            Base64.getEncoder().encodeToString(result.envelopeState.toByteArray()),
+                            null
+                        )
+                    )
+                }
+            }.fold(
+                onSuccess = {
+                    log.info("Successfully processed batch $index of ${chunkedFragResult.size}")
+                },
+                onFailure = {
+                    errors.add(it)
+                }
+            )
+        }
```

Comment:
hmm, yes, although that might be interesting to throw as returning actual objects might be most helpful for the caller to parse. Additionally, since you suggested limiting the error as well then we wont return them all. I can look into this as we use this endpoint more.